### ISSUE TITLE
[Feature]: Nuget SDK Resolver support for wildcard characters

### ISSUE THREAD

**Issue by dagasparmsft** [NuGet/Home#11783]
### NuGet Product(s) Involved

NuGet SDK

### The Elevator Pitch

We are creating a new SDK that partners will consume and will have consistent changes to the package. We want partners to be able to consume the latest version of the SDK and always build with the latest be being able to use wildcards such as 

```JSON
{
  "msbuild-sdks": {
    "Microsoft.Build.Traversal": "3.0.2",
    "Microsoft.Build.NoTargets": "2.0.1",
    "Microsoft.Substrate.Euclid.Sdk.Python": "1.0.*"
  },
  "sdk": {
  "version": "3.1.416"
  }
}
```

**Comment by jeffkl** [comment]
Duplicate of https://github.com/dotnet/msbuild/issues/3099 (I'm not sure why that one exists over on the MSBuild repo)

Thank you for your interest in this topic.  At this time, we have no plans to support wildcards for MSBuild project SDK versions. This is because project SDKs are resolved as MSBuild is evaluating a project.  When you load a large solution in Visual Studio, projects are evaluated in parallel and can cause UI delays if it doesn't happen fast.  Wildcards for NuGet package versions require NuGet to contact all of the configured feeds to resolve the version and if its already on disk, it doesn't download the package.  The HTTP cache keeps NuGet from making the network call for every single invocation, but the cache has a lifetime of I think 15 minutes.  

Large solutions would be greatly affected by this and it would make https://github.com/dotnet/msbuild/issues/4025 even worse.  What I suggest instead is an automated system that rolls out new versions of package like [dependabot](https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/).  This allows repos owners to see changes happening to their code base, roll back breaking changes, and service older branches that may or may not work with the "latest" version.

**Comment by nkolev92** [comment]
Closing per @jeffkl
